tests/utils/test_fmrisim.py

```diff
@@ -592,6 +592,22 @@ def test_generate_noise():
                        )
 
 
+def test_generate_noise_spatial():
+
+    # Set up the inputs
+    dimensions = np.array([10, 5, 10])
+    mask = np.ones(dimensions)
+    vol = sim._generate_noise_spatial(dimensions, mask)
+
+    assert np.all(vol.shape == dimensions), 'Volume incorrect shape'
+
+    # Check that the variance is similar across dimensions
+    x_var = (np.std(vol, 0) ** 2).mean()
+    y_var = (np.std(vol, 1) ** 2).mean()
+
+    assert abs(x_var - y_var) < y_var * 2, 'Variance is incorrect across dim'
```

Comment:
I kind of feel this criterion is too loose but I also don't know what a good test is.  I would think something like a F-test would be good to test for variance (https://www.itl.nist.gov/div898/handbook/eda/section3/eda359.htm) but you might need more samples (larger grids) to do such test, which makes the computation slow. Anyway, I just feel that testing a ratio of variance would be more appropriate.
An alternative way which may be a bit complex, is to test the degree of smoothness across the 3 dimensions.You may calculate fft of `vol`, and then calculate the indices for each grid in fft(vol) by your `_fftIndgen` function, and calculate the weighted average of indices with the weight being the squared magnitude of the fft, and compare this number with sigma^2. This way, you can check if the spatial smoothness matches the desired one. But of course it still is not checking variance.

Reply:
This is a great suggestion. I can use _calc_fwhm for this purpose. I will update the code accordingly.